### PR TITLE
[tests-only] added tests for COPY request endpoints with body

### PR DESCRIPTION
**Pull request by kiranadh1452**
## Description
This PR adds the test for the `COPY` request to endpoints with body for `core` and `oCIS` backend.

## Related Issue

- Part of https://github.com/owncloud/core/issues/40144

## Motivation and Context

https://datatracker.ietf.org/doc/html/rfc4918#section-8.4
If a body is sent, then the server is supposed to return HTTP status 415 (Unsupported Media Type).

## How Has This Been Tested?

- locally
- CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [X] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiSharees-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/36743/64

File: tests/acceptance/features/apiAuthWebDav/webDavCOPYAuth.feature [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+    Then the HTTP status code of responses on all endpoints should be "415"
+
+ @skipOnOcV10 @personalSpace
+  Scenario: send COPY requests to webDav endpoints with body as normal user using the spaces WebDAV API
+    When user "Alice" requests these endpoints with "COPY" including body "doesnotmatter" about user "Alice"
+      | endpoint                                           |
+      | /remote.php/dav/spaces/%spaceid%/textfile0.txt     |
+      | /remote.php/dav/spaces/%spaceid%/PARENT            |
+      | /remote.php/dav/spaces/%spaceid%/PARENT/parent.txt |
+    Then the HTTP status code of responses on all endpoints should be "415"
```
Comment (amrita-shrestha) [inline tests/acceptance/features/apiAuthWebDav/webDavCOPYAuth.feature:182]:
```suggestion
    Then the HTTP status code of responses on all endpoints should be "415"
    
```
add one extra line at the end of file

File: tests/acceptance/features/apiAuthWebDav/webDavCOPYAuthOC10Issue40144.feature [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    Then the HTTP status code of responses on all endpoints should be "400"
+  
+
+  Scenario: send COPY requests to webDav endpoints with body as normal user using the spaces WebDAV API
+    When user "Alice" requests these endpoints with "COPY" including body "doesnotmatter" about user "Alice"
+      | endpoint                                           |
+      | /remote.php/dav/spaces/%spaceid%/textfile0.txt     |
+      | /remote.php/dav/spaces/%spaceid%/PARENT            |
+      | /remote.php/dav/spaces/%spaceid%/PARENT/parent.txt |
+    Then the HTTP status code of responses on all endpoints should be "415"
```
Comment (amrita-shrestha) [inline tests/acceptance/features/apiAuthWebDav/webDavCOPYAuthOC10Issue40144.feature:33]:
```suggestion
    Then the HTTP status code of responses on all endpoints should be "415"
    
```

File: tests/acceptance/features/apiAuthWebDav/webDavCOPYAuth.feature [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+   When user "Brian" requests these endpoints with "COPY" about user "Alice"
+     | endpoint                                           |
+     | /remote.php/dav/spaces/%spaceid%/textfile0.txt     |
+     | /remote.php/dav/spaces/%spaceid%/PARENT            |
+     | /remote.php/dav/spaces/%spaceid%/PARENT/parent.txt |
+   Then the HTTP status code of responses on all endpoints should be "403"
+
+
+  Scenario: send COPY requests to webDav endpoints using invalid username but correct password
+    When user "usero" requests these endpoints with "COPY" using the password of user "Alice"
```
Comment (amrita-shrestha) [inline tests/acceptance/features/apiAuthWebDav/webDavCOPYAuth.feature:76]:
```suggestion
    When user "usero" requests these endpoints with "COPY" using the password of user "Alice"
```
`user0` has not been created

File: tests/acceptance/features/apiAuthWebDav/webDavCOPYAuth.feature [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+      | /remote.php/webdav/textfile0.txt                   |
+      | /remote.php/dav/files/%username%/textfile0.txt     |
+      | /remote.php/webdav/PARENT                          |
+      | /remote.php/dav/files/%username%/PARENT            |
+      | /remote.php/dav/files/%username%/PARENT/parent.txt |
+    Then the HTTP status code of responses on all endpoints should be "401"
+
+  @skipOnOcV10 @personalSpace
+  Scenario: send COPY requests to webDav endpoints using invalid username but correct password using the spaces WebDAV API
+    When user "usero" requests these endpoints with "COPY" using the password of user "Alice"
```
Comment (amrita-shrestha) [inline tests/acceptance/features/apiAuthWebDav/webDavCOPYAuth.feature:87]:
```suggestion
    When user "usero" requests these endpoints with "COPY" using the password of user "Alice"
```
`user0` has not been created

File: tests/acceptance/features/apiAuthWebDav/webDavCOPYAuth.feature (hunk 1 above)
Comment (amrita-shrestha) [inline tests/acceptance/features/apiAuthWebDav/webDavCOPYAuth.feature:182]: this scenario is duplicated in webDavCOPYAuthOC10Issue40144.feature without any difference. Could you explain?

File: tests/acceptance/features/apiAuthWebDav/webDavCOPYAuth.feature [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+      | username |
+      | Alice    |
+      | Brian    |
+    And user "Alice" has uploaded file with content "some data" to "/textfile0.txt"
+    And user "Alice" has created folder "/PARENT"
+    And user "Alice" has created folder "/FOLDER"
+    And user "Alice" has uploaded file with content "some data" to "/PARENT/parent.txt"
+
+  @smokeTest
+  @skipOnBruteForceProtection @issue-brute_force_protection-112
```
Comment (amrita-shrestha) [inline tests/acceptance/features/apiAuthWebDav/webDavCOPYAuth.feature:15]:
```suggestion
  @smokeTest @skipOnBruteForceProtection @issue-brute_force_protection-112
```

File: tests/acceptance/features/apiAuthWebDav/webDavCOPYAuth.feature [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+  Scenario: send COPY requests to webDav endpoints as normal user with wrong password using the spaces WebDAV API
+    When user "Alice" requests these endpoints with "COPY" using password "invalid" about user "Alice"
+      | endpoint                                           |
+      | /remote.php/dav/spaces/%spaceid%/textfile0.txt     |
+      | /remote.php/dav/spaces/%spaceid%/PARENT            |
+      | /remote.php/dav/spaces/%spaceid%/PARENT/parent.txt |
+    Then the HTTP status code of responses on all endpoints should be "401"
+
+  @smokeTest
+  @skipOnBruteForceProtection @issue-brute_force_protection-112
```
Comment (amrita-shrestha) [inline tests/acceptance/features/apiAuthWebDav/webDavCOPYAuth.feature:36]:
```suggestion
  @smokeTest @skipOnBruteForceProtection @issue-brute_force_protection-112
```

File: tests/acceptance/features/apiAuthWebDav/webDavCOPYAuth.feature [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+
+  @skipOnOcV10 @personalSpace
+  Scenario: send COPY requests to webDav endpoints using invalid username but correct password using the spaces WebDAV API
+    When user "usero" requests these endpoints with "COPY" using the password of user "Alice"
+      | endpoint                                           |
+      | /remote.php/dav/spaces/%spaceid%/textfile0.txt     |
+      | /remote.php/dav/spaces/%spaceid%/PARENT            |
+      | /remote.php/dav/spaces/%spaceid%/PARENT/parent.txt |
+    Then the HTTP status code of responses on all endpoints should be "401"
+
```
Comment (amrita-shrestha) [inline tests/acceptance/features/apiAuthWebDav/webDavCOPYAuth.feature:93]:
```suggestion


```

File: tests/acceptance/features/apiAuthWebDav/webDavCOPYAuth.feature [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+  Scenario: send COPY requests to webDav endpoints using valid password and username of different user using the spaces WebDAV API
+    When user "Brian" requests these endpoints with "COPY" using the password of user "Alice"
+      | endpoint                                           |
+      | /remote.php/dav/spaces/%spaceid%/textfile0.txt     |
+      | /remote.php/dav/spaces/%spaceid%/PARENT            |
+      | /remote.php/dav/spaces/%spaceid%/PARENT/parent.txt |
+    Then the HTTP status code of responses on all endpoints should be "401"
+
+  @smokeTest
+  @skipOnBruteForceProtection @issue-brute_force_protection-112
```
Comment (amrita-shrestha) [inline tests/acceptance/features/apiAuthWebDav/webDavCOPYAuth.feature:114]:
```suggestion
  @smokeTest @skipOnBruteForceProtection @issue-brute_force_protection-112
```

File: tests/acceptance/features/apiAuthWebDav/webDavCOPYAuthOC10Issue40144.feature [hunk 9]
```diff
@@ -0,0 +1,33 @@
+@api @notToImplementOnOCIS @skipOnOcis
```
Comment (SwikritiT) [inline tests/acceptance/features/apiAuthWebDav/webDavCOPYAuthOC10Issue40144.feature:1]:
```suggestion
@api @skipOnOcis
```
Only one tag is enough they both do the same thing

File: tests/acceptance/features/apiAuthWebDav/webDavCOPYAuthOC10Issue40144.feature [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+    Then the HTTP status code of responses on all endpoints should be "403"
+  
+
+  Scenario: send COPY requests to webDav endpoints with body as normal user using the spaces WebDAV API
+    When user "Alice" requests these endpoints with "COPY" including body "doesnotmatter" about user "Alice"
+      | endpoint                                           |
+      | /remote.php/dav/spaces/%spaceid%/textfile0.txt     |
+      | /remote.php/dav/spaces/%spaceid%/PARENT            |
+      | /remote.php/dav/spaces/%spaceid%/PARENT/parent.txt |
+    Then the HTTP status code of responses on all endpoints should be "403"
```
Comment (grgprarup) [inline tests/acceptance/features/apiAuthWebDav/webDavCOPYAuthOC10Issue40144.feature:33]:
```suggestion
```
This scenario is for spaces. It only works for OCIS.

File: tests/acceptance/features/apiAuthWebDav/webDavCOPYAuth.feature [hunk 11]
```diff
@@ -0,0 +1,180 @@
+@api
+Feature: COPY file/folder
+
```
Comment (saw-jan) [inline tests/acceptance/features/apiAuthWebDav/webDavCOPYAuth.feature:3]: It is better to write a user story as well

File: tests/acceptance/features/apiAuthWebDav/webDavCOPYAuthOC10Issue40144.feature [hunk 12]
```diff
@@ -0,0 +1,24 @@
+@api @skipOnOcis
+Feature: COPY file/folder
+
```
Comment (saw-jan) [inline tests/acceptance/features/apiAuthWebDav/webDavCOPYAuthOC10Issue40144.feature:3]: recommended to add user story

**Review by SagarGi — COMMENTED** [review]
LGTM :+1:  (good to have tests for COPY req as well) Lets make the CI green

**Review by phil-davis — APPROVED** [review]

**Pull request merged**